scripts/azureml-assets/azureml/assets/model/model_utils.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Model utils Operations Class."""
+
+import os
+import azureml.assets as assets
+from azure.identity import DefaultAzureCredential
+from azure.ai.ml import load_model
+from azure.ai.ml._utils._registry_utils import get_asset_body_for_registry_storage, get_registry_client
+from azureml.assets.util import logger
+from azureml.assets.config import PathType
+from azureml.assets.model.download_utils import copy_azure_artifacts, download_git_model
+from pathlib import Path
+
+
+class RegistryUtils:
+    """Registry utils."""
+
+    RETRY_COUNT = 3
+
+    def get_registry_data_reference(registry_name: str, model_name: str, model_version: str):
+        """Fetch data reference for asset in the registry."""
+        cnt, blob_uri, sas_uri = 0, None, None
+        asset_id = f"azureml://registries/{registry_name}/models/{model_name}/versions/{model_version}"
+        logger.print(f"getting data reference for asset {asset_id}")
+        while cnt < RegistryUtils.RETRY_COUNT:
+            try:
+                response = RegistryUtils._get_temp_data_ref(registry_name, model_name, model_version)
+                blob_uri = response.blob_reference_for_consumption.blob_uri
+                sas_uri = response.blob_reference_for_consumption.credential.additional_properties["sasUri"]
+                if not blob_uri or not sas_uri:
+                    raise Exception("Error in fetching BLOB or SAS URI")
+                break
+            except Exception as e:
+                cnt += 1
+                logger.log_error(f"Exception in fetching data reference. Retry count {cnt}. Error: {e}")
+        if not blob_uri or not sas_uri:
+            raise Exception(f"Unable to fetch data reference for asset {asset_id}")
+        return blob_uri, sas_uri
+
+    def _get_temp_data_ref(registry_name, model_name, model_version):
+        try:
+            credential = DefaultAzureCredential()
```

Comment:
Please consider using `AzureCliCredential` instead, rather than . No other credential should be available during release. See https://learn.microsoft.com/en-us/python/api/overview/azure/identity-readme?view=azure-python#authenticate-via-the-azure-cli

Reply:
Done

Reply:
But it says both AzureCliCredential and DefaultAzureCredential is available during CLI login. Isn't that the case during release scenario?

Reply:
DefaultAzureCredential tries a lot of login methods and stops at the first one that succeeds. I'd rather we just go for the correct one in the first place, which AzureCliCredential provides.

Reply:
yes true. Updated